dbt/adapters/spark/python_submissions.py

```diff
@@ -0,0 +1,282 @@
+
+    def get_timeout(self):
+        timeout = self.parsed_model["config"].get("timeout", 60 * 60 * 24)
+        if timeout <= 0:
+            raise ValueError("Timeout must be a positive integer")
+        return timeout
+
+    def check_credentials(self, credentials):
+        raise NotImplementedError(
+            "Overwrite this method to check specific requirement for current submission method"
+        )
+
+    def submit(self, compiled_code):
+        raise NotImplementedError(
+            "BasePythonJobHelper is an abstract class and you should implement submit method."
+        )
+
+    def polling(
+        self,
+        status_func,
+        status_func_kwargs,
+        get_state_func,
+        terminal_states,
+        expected_end_state,
+        get_state_msg_func,
+    ):
+        state = None
+        start = time.time()
+        exceeded_timeout = False
+        response = {}
+        while state not in terminal_states:
+            if time.time() - start > self.timeout:
+                exceeded_timeout = True
+                break
+            # TODO should we do exponential backoff?
+            time.sleep(self.polling_interval)
+            response = status_func(**status_func_kwargs)
+            state = get_state_func(response)
+        if exceeded_timeout:
+            raise dbt.exceptions.RuntimeException("python model run timed out")
+        if state != expected_end_state:
+            raise dbt.exceptions.RuntimeException(
+                "python model run ended in state"
+                f"{state} with state_message\n{get_state_msg_func(response)}"
+            )
+        return response
+
+
+class DBNotebookPythonJobHelper(BasePythonJobHelper):
+    def __init__(self, parsed_model, credentials):
+        super().__init__(parsed_model, credentials)
+        self.auth_header = {"Authorization": f"Bearer {self.credentials.token}"}
+
+    def check_credentials(self, credentials):
+        if not credentials.user:
+            raise ValueError("Databricks user is required for notebook submission method.")
+
+    def _create_work_dir(self, path):
+        response = requests.post(
+            f"https://{self.credentials.host}/api/2.0/workspace/mkdirs",
+            headers=self.auth_header,
+            json={
+                "path": path,
+            },
+        )
+        if response.status_code != 200:
+            raise dbt.exceptions.RuntimeException(
+                f"Error creating work_dir for python notebooks\n {response.content!r}"
+            )
+
+    def _upload_notebook(self, path, compiled_code):
+        b64_encoded_content = base64.b64encode(compiled_code.encode()).decode()
+        response = requests.post(
+            f"https://{self.credentials.host}/api/2.0/workspace/import",
+            headers=self.auth_header,
+            json={
+                "path": path,
+                "content": b64_encoded_content,
+                "language": "PYTHON",
```

Comment:
not sure it matters but looks like 'language' is uppercase here and lowercase elsewhere, maybe put this in a static variable?

Reply:
I put the rest into a static variable but this one needs to be upper cased as the API is a different one and actually require uppercase.